sql/upsert.go

```diff
@@ -102,6 +87,23 @@ func (p *planner) makeUpsertHelper(
 		evalExprs = append(evalExprs, normExpr)
 	}
 
+	allExprsIdentity := true
+	for i, expr := range evalExprs {
+		// analyzeExpr above has normalized all direct column names to ColumnItems.
+		c, ok := expr.(*parser.ColumnItem)
+		if !ok {
+			allExprsIdentity = false
+			break
+		}
+		// FIXME(knz) check other db name is OK here.
```

Comment:
@paperstreet @dt I'd like your feedback here. The code prior to this point was not checking the database name prefix to these expressions. Was this on purpose, or do we need to check the db name is not set?